service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/RetrieveDestinationStorageAccountResourceIdStep.java

```diff
@@ -0,0 +1,127 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.azure.container;
+
+import bio.terra.common.iam.BearerToken;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneDeployedResource;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.flight.clone.azure.container.exception.InvalidStorageAccountException;
+import bio.terra.workspace.service.resource.model.WsmResourceFamily;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Attempts to fetch the storage account ID for a new container in the destination workspace
+ * following this process:
+ *
+ * <ol>
+ *   <li>Checks the destination workspace for a single storage account; if more than one is present
+ *       we are in an unsupported state and fail out.
+ *   <li>If no storage accounts are present, attempts to check the owning Landing Zone. If present,
+ *       returns the storage account from the LZ, otherwise we fail out.
+ * </ol>
+ */
+public class RetrieveDestinationStorageAccountResourceIdStep implements Step {
```

Comment:
I feel like this logic is duplicated in the VerifyAzureStorageContainerCanBeCreated step.https://github.com/DataBiosphere/terra-workspace-manager/blob/688c3f42db57fcb1206f2a3c3a45ff6d1d12f85e/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java#L140